### ISSUE TITLE
Rotary not accepted

### ISSUE THREAD

**Issue by Shane-Lester** [tjanson/PiSwitch#7]
I get an error when I send 'rotary'
ERROR [piswitch.js] bad input (type: rotary)

I don't understand as rotary is a defined type

Shane

**Comment by tjanson** [comment]
This means the code you tried to send did not match the input format for
rotary. Rotary simply consists of two digits between 1 and 4. What code did
you try to send?

**Comment by Shane-Lester** [comment]
OK - that's helpful. I was doing A1. 
I've now tried 11 and it works. Thanks

Shane

**Comment by Shane-Lester** [comment]
Thanks for the support

**Comment by Shane-Lester** [comment]
OK, 
With your help I've got it working. What I'd really like to do is run it from node-red. Do you know how I can do this?

Shane

**Comment by tjanson** [comment]
I really don't know much about Node-RED, but I believe you can run
arbitrary commands using the `exec` node. You can just use the example
script and execute it with `node`(.js) (which is probably exactly what you
did so far). If exporting the pin beforehand is cumbersome, you could run
`node`(.js) as root using `sudo` and change the `mode` in the `setup`
method accordingly (see the example script).

I hope this wasn't too confusing — silly Node-RED makes talking about
"node" ambiguous.

**Comment by Shane-Lester** [comment]
Thanks for your help. Will give it a whirl. 

All the best

Shane

**Comment by Shane-Lester** [comment]
Great- this is what I did
Changed the default to rotary
Changed the order args 3&4 so I can omit the device type and rely on the default. 
Used an exec node. Don't append (couldn't make this work)

The command is 
/usr/bin/sudo 

The additional info is
node <path to dir>/example.js 11 off

And it works

I have to use separate on and off nodes as append wouldn't work. Not a big deal though

Thanks

Shane
